### ISSUE TITLE
Count events in multiple relays

### ISSUE THREAD

**Issue by SnowCait** [nostr-protocol/nips#765]
## Problem

[NIP-45](https://github.com/nostr-protocol/nips/blob/master/45.md) COUNT can count events in a relay, but all events don't exist in one relay.

### Example

Example of follower counts with `3` followers.

| wss://a.relay.com/ |
| - |
| Me |
| Follower-A |
| Follower-B |

| wss://b.relay.com/ |
| - |
| Follower-B |
| Follower-C |

Requesting `["COUNT", "_", {"kinds": 3, "#p", ["<ME>"]}` to wss://a.relay.com/ returns `["COUNT", "_", {"count": 2}]`.  
Requesting `["COUNT", "_", {"kinds": 3, "#p", ["<ME>"]}` to wss://b.relay.com/ returns `["COUNT", "_", {"count": 2}]`.  
We can't merge the values of COUNT.

## Solution

Return `ids`.

If relays return `["COUNT", "_", {"count": 2, "ids":["<Follower-A>", "<Follower-B>"]}]` and `["COUNT", "_", {"count": 2, "ids":["<Follower-B>", "<Follower-C>"]}]`, we can merge `ids` and count them.

**Comment by alexgleason** [comment]
Interesting idea to return the IDs, but it seems like it would be a problem at a certain point.

![image](https://github.com/nostr-protocol/nips/assets/3639540/4d3efb3a-a8af-4b58-966a-97634ac019b0)

The better thing to do is to figure out the best relay to read the COUNT from. You could do that by getting the kind 10002 event for the author of the post before the COUNT request.

Proper clients will send their replies and likes to the relay(s) in the author's 10002 event, making it the most canonical source for stats.

**Comment by SnowCait** [comment]
I agree with that if we can force all clients to implement [NIP-65](https://github.com/nostr-protocol/nips/blob/master/65.md)...

**Comment by alexgleason** [comment]
NIP-05 can also have the relay. You could try kind 10002 fallback to NIP-05.

Of course you will need a fallback if you still can't find a good relay. I would suggest:

- Option 1: Get COUNT on a pool, sort the results, and then take the **median** value.
- Option 2: Keep events in your own database on the client, and then take the count from your own database.

**Comment by SnowCait** [comment]
It does not make sense for this problem to depend on optional client/user implementation, median value or own database.
Please read [NIP-45 Motivation](https://github.com/nostr-protocol/nips/blob/master/45.md#motivation).

**Comment by AsaiToshiya** [comment]
@SnowCait DVM is probably [NIP-90: Data Vending Machines](https://github.com/nostr-protocol/nips/blob/master/90.md).

**Comment by staab** [comment]
Got a count DVM merged into the data-vending-machines repo, see [kind 5400](https://www.data-vending-machines.org/ranges/54xx/)
